### DISCUSSION TITLE
How to  make cookie.get() work with static-adapter after build

### DISCUSSION THREAD

**Discussion by Mrbeyond** [sveltejs/kit#8653] [Q&A] [answered]
I'm able to get cookies value locally using cookie.get() method in +layout.sever and +page.server files.

But after production build with static-adapter, cookie.get  returns undefined even if the cookie can be seen on the browser.

Is this an expected behaviour or there's issue with the build?

Everything works perfectly locally.

**Comment by teemingc** [comment, marked as answer]
This is expected behaviour.

`adapter-static` prerenders your pages at build time, so your server functions are only run during build time, and not anymore after that. Hence, your cookie is never read after build.

With prerendered pages, the only dynamic values are those set in the browser such as during `onMount`.

You may want to read the cookie value in the browser (if it isn't http only), or not use `adapter-static`

**Reply by Mrbeyond** [reply]
@s3812497 Thanks for your response, please which other adapter is suitable for nginx server?

**Reply by teemingc** [reply]
You should be able to use `adapter-node` to run it with node installed on that server (sorry, I’m not familiar with nginx)

more about the adapter here:
https://kit.svelte.dev/docs/adapter-node

and a stackoverflow thread I found about nginx and sveltekit
https://stackoverflow.com/questions/70398935/how-to-deploy-a-svelte-kit-app-after-build-using-nginx-as-web-server/70434644

**Reply by Mrbeyond** [reply]
@s3812497 Thanks a lot.
